spectral_cube/spectral_cube.py

```diff
@@ -171,18 +171,22 @@ class BaseSpectralCube(BaseNDClass, MaskableArrayMixinClass,
                        HeaderMixinClass):
 
     def __init__(self, data, wcs, mask=None, meta=None, fill_value=np.nan,
-                 header=None, allow_huge_operations=False, wcs_tolerance=0.0):
+                 header=None, allow_huge_operations=False, wcs_tolerance=0.0,
+                 unit=None):
 
         # Deal with metadata first because it can affect data reading
         self._meta = meta or {}
 
         # must extract unit from data before stripping it
-        if 'BUNIT' in self._meta:
-            self._unit = cube_utils.convert_bunit(self._meta["BUNIT"])
-        elif hasattr(data, 'unit'):
-            self._unit = data.unit
+        if unit is None:
+            if 'BUNIT' in self._meta:
+                self._unit = cube_utils.convert_bunit(self._meta["BUNIT"])
+            elif hasattr(data, 'unit'):
+                self._unit = data.unit
```

Comment:
```suggestion
            elif hasattr(data, 'unit'):
                self._unit = data.unit
            elif hasattr(data, '_meta'):
                self._unit = getattr(data._meta, 'unit', None)
```
might be worth a shot